vocabulary-draft/source-code-version-control.md

```diff
@@ -0,0 +1,27 @@
+# Source Code Version Control Events
+
+This phase includes events related to changes in Source Code repositories that are relevant from a Continuous Delivery perspective.
+
+
+## Repository Events
+
+These events are related to Source Code repositories
+- **Repository Created Event**: a new Source Code Repository was created to host source code for a project
+- **Repository Modified Event**: a Source Code Repository modified some of its attributes, like location, or owner
+- **Repository Deleted Event**: a Source Code Repository was deleted and it is not longer available
+- **Branch Created Event**: a Branch inside the Repository was created 
+- **Branch Deleted Event**: a Branch inside the Repository was deleted
+- **Change Committed**: a change is commited to a branch
+- **Change Approved**: a change request is created by a user, that requires review and approval before being applied to an existing branch inside the repository
```

Comment:
If a review requires several approvals would we send an event for each? For example

- Code review 
- CI engine verifications
- Code guardian approval

Reply:
I think that `Change Approved` was trying to reflect that. We can capture different actors for example the `Approver` with data inside the event itself. Again I was trying to focus on the minimal set of metadata that we need to get started. But we can definitely add more to it.

Reply:
So your idea was to send one event when all approvers have given their content? If we go with one event for each approval we could:

* Capture code +1 but CI engine -1
* Know the time between submission and guardian approval

Or was there some other idea?

Reply:
@m-linner-ericsson we can send an event per approver and another event when all the expected approvers are done. I think that we can use an attribute to identify if these approvers are people or automated software.